ARCs/arc-0018.md

```diff
@@ -125,13 +131,35 @@ The `from` parameter is the address for the account the ASA is transferred from.
 
 The `to` parameter is the address for the account the ASA is transferred to.
 
-The `royalty_collector` parameter is the address for the account that collects the royalty and takes care of any redistribution logic (if any). 
+The `royalty_receiver` parameter is the address for the account that collects the royalty and takes care of any redistribution logic (if any). 
 
 The `payment` parameter is a reference to the transaction that is transferring some asset (ASA or Algos) from the buyer to the Application Address. 
 
-Upon each transfer the the Royalty Enforcer MUST assert that `AuthAddr == AssetSender` for the Seller (`from` parameter) and `AuthAddr == AssetReceiver` for the Buyer (`to` parameter), in other words Seller and Buyer accounts have not been Rekeyed. Exceptions can be handled with `Move` method.
+Upon each transfer the the Royalty Enforcer MUST assert that `AuthAddr == AssetSender` for the Seller (`from` parameter) and `AuthAddr == AssetReceiver` for the Buyer (`to` parameter). In other words Seller and Buyer accounts not have been Rekeyed. Exceptions can be handled with `Move` method.
+
+> The rational for this check is ensuring that `Rekeying` is not exploited to transfer the `royalty_asset` ownership circumventing the Royalty Policy. Two parties involved in the trade could create a Group Transactions in which: 1. Seller rekeys the `from` to the Buyer `to`, 2. Buyer sends payment to `other_seller_account`. Note that nothing can prevent the users to execute such trade but the Royalty Enforcer tries to dissuade this behavior: if such trade happens, the Receiver will no longer be able to re-sell the Royalty ASA in the regular way.
+
+
+#### Offer:
+
+    offer(royalty_asset: asset, authorized_address: account)
```

Comment:
See this [comment](https://github.com/algorandfoundation/ARCs/pull/70#discussion_r838336925). I think we should add `RoyaltyAssetAmount` here too.